crates/minecraft_packets/src/play/set_player_position.rs

```diff
@@ -0,0 +1,15 @@
+use minecraft_protocol::prelude::*;
+
+#[derive(PacketIn)]
+pub struct SetPlayerPositionPacket {
+    pub x: f64,
+    pub feet_y: f64,
+    pub z: f64,
+    pub flags: u8,
```

Comment:
Before 1.21.4 the flags is a boolean

> On Ground 	Boolean	True if the client is on the ground, false otherwise.

```rust
#[pvn(769..)]
pub v1_21_4_flags: u8,
#[pvn(..769)]
pub on_ground: bool,
```